dll/win32/advapi32/reg/reg.c

```diff
@@ -2327,19 +2290,25 @@ RegDeleteValueA(HKEY hKey,
         return RtlNtStatusToDosError(Status);
     }
 
-    RtlCreateUnicodeStringFromAsciiz(&ValueName, lpValueName);
-    Status = NtDeleteValueKey(KeyHandle,
-                              &ValueName);
-    RtlFreeUnicodeString (&ValueName);
-
-    ClosePredefKey(KeyHandle);
-
-    if (!NT_SUCCESS(Status))
+    if (!RtlCreateUnicodeStringFromAsciiz(&ValueName, lpValueName))
     {
-        return RtlNtStatusToDosError(Status);
+        ClosePredefKey(KeyHandle);
+        return ERROR_NOT_ENOUGH_MEMORY;
     }
 
-    return ERROR_SUCCESS;
+    if (IsHKCRKey(KeyHandle))
+    {
+        ErrorCode = DeleteHKCRValue(KeyHandle, &ValueName);
+    }
+    else
+    {
+        Status = NtDeleteValueKey(KeyHandle, &ValueName);
+        if (!NT_SUCCESS(Status))
+            ErrorCode = RtlNtStatusToDosError(Status);
+    }
+    RtlFreeUnicodeString (&ValueName);
```

Comment:
```suggestion
    RtlFreeUnicodeString(&ValueName);
```

Reply:
This was existing code but I can fix it.